### PR TITLE
Add sign-up page

### PR DESCRIPTION
**Pull request by miguelbrandao**
Adds a sign-up page.
Updates the README with instructions to test the page in development.

After sign-up, the user is logged-in and redirected to the main page

### CODE REVIEW

**Comment by apedrob** [comment]
The folder `src` should have the pages, components, containers ou even style subfolders. And if `Link.js` is a component, it needs to be inside of components folder

File: components/NavBar.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ import Box from '@material-ui/core/Box';
 import Toolbar from '@material-ui/core/Toolbar';
 import Grid from '@material-ui/core/Grid';
 import Typography from '@material-ui/core/Typography';
+import Link from '../src/Link';
```
Comment (apedrob) [inline components/NavBar.js]: ⚠️ folder

File: components/NavBar.js [hunk 2]
```diff
@@ -27,17 +28,27 @@ const Entry = ({ name }) => (
...
   );
 
+// The signup link is not intended to stay in the navbar;
+// it's there for development only. It may be incorporated into 
+// the signin page in the future if deemed apropriate
 const NavBar = () => (
   <StyledBox>
     <StyledToolBar>
-      <Entry name="Test1" />
+      <Link href="/">
```
Comment (apedrob) [inline components/NavBar.js]: Don't navigate between pages with `href`

File: components/NavBar.js [hunk 3]
```diff
@@ -27,17 +28,27 @@ const Entry = ({ name }) => (
...
+      </Link>
       <Grid
           container
           direction="row-reverse"
           justify="flex-start"
           alignItems="center"
           spacing={1}
       >
+        <Grid item>
+          <Link href="/signup">
```
Comment (apedrob) [inline components/NavBar.js]: Same here ⚠️

File: pages/signup.js [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+}
+
+export default function SignUp() {
+  const classes = useStyles();
+  
+  const router = useRouter()
+  let UUID = router.query.id;
+  //console.log(UUID);
+
+  const [errorMsg, setErrorMsg] = React.useState('');
```
Comment (apedrob) [inline pages/signup.js]: Let's use `Redux` to handle states

File: pages/signup.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    <Container component="main" maxWidth="xs">
+    <CssBaseline />
+    <div className={classes.paper}>
+      <Avatar className={classes.avatar}>
+      <LockOutlinedIcon />
+      </Avatar>
+      <Typography component="h1" variant="h5">
+      Sign up
+      </Typography>
+      <form className={classes.form} noValidate onSubmit={handleSubmit}>
```
Comment (apedrob) [inline pages/signup.js]: This should be a Form component

File: src/Link.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import { useRouter } from 'next/router';
+import NextLink from 'next/link';
+import MuiLink from '@material-ui/core/Link';
+
+const NextComposed = React.forwardRef(function NextComposed(props, ref) {
+  const { as, href, prefetch, ...other } = props;
+
+  return (
+    <NextLink href={href} prefetch={prefetch} as={as}>
+      <a ref={ref} {...other} />
```
Comment (apedrob) [inline src/Link.js]: we should describe all the `props`

**Comment by apedrob** [comment]
And what is happing with `now`, I want to see the preview 👀

File: components/NavBar.js (hunk 2 above)
Comment (miguelbrandao) [reply components/NavBar.js]: But that's the way [NextJs' tutorial](https://nextjs.org/learn/basics/navigate-between-pages/using-link) specify. Am I missing something? How should it be done?

File: pages/signup.js (hunk 4 above)
Comment (miguelbrandao) [reply pages/signup.js]: @herulume advised us to stay away from it tho. Maybe he could chime in and discuss this?

**Comment by miguelbrandao** [comment]
@apedrob this works, but it's very unorganized and I'm sure there's a better way to arrange this. Do you have any suggestions?

**Comment by miguelbrandao** [comment]
This will be in another PR

**Pull request closed without merging**